### ISSUE TITLE
C# float doesn't correspond to the correct float type of GDScript implementation

### ISSUE THREAD

**Issue by Coder2195Text** [godotengine/godot#64397]
### Godot version

3.4.4stable

### System information

Windows 11, OpenGL ES 3.0 Renderer: Intel(R) UHD Graphics 620

### Issue description

This is the definition of a Vector2 in GDScript.
![image](https://user-images.githubusercontent.com/63069499/184544646-1c4eb2fb-73c6-480c-ba9f-402a43c0da86.png)
In turn, it defines float as:
![image](https://user-images.githubusercontent.com/63069499/184544669-1dc367d6-57ec-43d0-ae8e-a02f221383e8.png)
It says 64bit, or a double in C++
Now, C++ doubles in C# are also doubles.
When I look at the type definition of the Vector2 implementation in C#, instead of seeing a double, I see float
![image](https://user-images.githubusercontent.com/63069499/184544746-d8aa5414-8551-4465-ac50-bfd19fd9e0ed.png)
This contradicts the 64 bit floating types that are implementation in the GDScript version of Vector

### Steps to reproduce

Look in C# Vector2 type defs

**Comment by Calinou** [comment]
> When I look at the type definition of the Vector2 implementation in C#, instead of seeing a double, I see float

This is also the case in GDScript. Only primitive floats are 64-bit in GDScript; Vector components are 32-bit.

**Comment by Coder2195Text** [comment]
is it possible to use 64bit floats vectors in Godot3? Because I actually need this for a game, since it should strech as far like minecraft

**Comment by Calinou** [comment]
> is it possible to use 64bit floats vectors in Godot3?

No, as the functionality was never brought to an usable state (even though the build-time define `REAL_T_IS_DOUBLE` technically exists).

> Because I actually need this for a game, since it should strech as far like minecraft

How large is your game world? Judging by how [Minetest](https://minetest.net) handles this, it should be possible to have a 64km × 64km × 64km world without using double-precision floats.

**Comment by Coder2195Text** [comment]
Im implementing it via a 2d world, I expect 1000000*64 pixels wide

**Comment by Sythelux** [comment]
> is it possible to use 64bit floats vectors in Godot3? Because I actually need this for a game, since it should strech as far like minecraft

Instead of using 64bit float, look up "floating origin" on the web. It is a pattern to work with infinite big terrain. I use this for a digital twin simulation I do with Godot. It is probably the better alternative, as using double will immediately give you a performance penalty and the pattern will give you a performance improvement.

But we can move that to discussion. The important part is, that GDScript is working as intended on this matter?
